File: Лаба2/Задание 1 вариант 3/main.swift
import Foundation

func checkP(_ x: Int) -> Int {
    var b = 0
    var z: Int
    for i in stride(from: x, to: 1, by: -1) {
        z = 1
        for j in 2...Int(sqrt(Double(i))) {
            if i % j == 0 {
                z = 0
                break
            }
        }
        if z == 1 {
            b = i
            break
        }
    }
    for i in x... {
        z = 1
        for j in 2...Int(sqrt(Double(i))) {
            if i % j == 0 {
                z = 0
                break
            }
        }
        if z == 1 {
            if x - b > i - x {
                b = i
            }
            break
        }
    }
    return b
}

if let input = readLine(), let x = Int(input) {
    print(abs(x - checkP(x)))
} else {
    print("Invalid input")
}

File: Лаба2/Задание 1 вариант 3/main.rs
use std::io;

fn check_p(x: i32) -> i32 {
    let mut b = 0;
    let mut z;
    for i in (2..=x).rev() {
        z = 1;
        for j in 2..=((i as f64).sqrt() as i32) {
            if i % j == 0 {
                z = 0;
                break;
            }
        }
        if z == 1 {
            b = i;
            break;
        }
    }
    for i in x.. {
        z = 1;
        for j in 2..=((i as f64).sqrt() as i32) {
            if i % j == 0 {
                z = 0;
                break;
            }
        }
        if z == 1 {
            if x - b > i - x {
                b = i;
            }
            break;
        }
    }
    b
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let x: i32 = input.trim().parse().expect("Please enter a valid number");
    println!("{}", (x - check_p(x)).abs());
}